lib/dialects/mssql/transaction.js

```diff
@@ -60,43 +60,47 @@ module.exports = class Transaction_MSSQL extends Transaction {
   // Acquire a connection and create a disposer - either using the one passed
   // via config or getting one off the client. The disposer will be called once
   // the original promise is marked completed.
-  acquireConnection(_client, config, _txid) {
-    const t = this;
+  acquireConnection(config, cb) {
     const configConnection = config && config.connection;
     return new Bluebird((resolve, reject) => {
       try {
         resolve(
-          (t.outerTx ? t.outerTx.conn : null) ||
+          (this.outerTx ? this.outerTx.conn : null) ||
             configConnection ||
-            t.client.acquireConnection()
+            this.client.acquireConnection()
         );
       } catch (e) {
         reject(e);
       }
     })
-      .then(function(conn) {
-        conn.__knexTxId = t.txid;
-        if (!t.outerTx) {
-          t.conn = conn;
+      .then((conn) => {
```

Comment:
would be awesome to convert those `then` chains into async/await, but separate PR would be better for that.